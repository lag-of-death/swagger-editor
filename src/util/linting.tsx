import { Spectral } from "@stoplight/spectral";
import { oas2Functions, oas2Rules } from "@stoplight/spectral/rulesets/oas2";

// usage: diagnose(require('../../spec/petstore.oas2.json'))
export function diagnose(spec: object) {
  const spectral = new Spectral();
  spectral.addFunctions(oas2Functions());
  spectral.addRules(oas2Rules());
  return spectral.run(spec, {resolvedTarget: spec}).results;
}
