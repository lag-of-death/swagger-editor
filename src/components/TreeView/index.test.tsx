import { shallow } from "enzyme";
import * as React from "react";
import petstore = require("../../../spec/petstore.oas2.json");
import { diagnose } from "../../util/linting";
import { TreeView } from "./index";

describe("TreeView", () => {
  test("rendering", () => {
    const component = shallow(
      <TreeView spec={petstore}
                diagnostics={diagnose(petstore)}/>,
    );

    expect(component).toMatchSnapshot();
  });
});
