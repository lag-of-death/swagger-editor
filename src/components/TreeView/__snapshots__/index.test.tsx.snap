// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView rendering 1`] = `
<ul>
  <Leaf
    key="swagger"
    labelKey="swagger"
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="info"
    labelKey="info"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Leaf
    key="host"
    labelKey="host"
    renderIssuesIndicators={[Function]}
  />
  <Leaf
    key="basePath"
    labelKey="basePath"
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="tags"
    labelKey="tags"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="schemes"
    labelKey="schemes"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="paths"
    labelKey="paths"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="securityDefinitions"
    labelKey="securityDefinitions"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="definitions"
    labelKey="definitions"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="externalDocs"
    labelKey="externalDocs"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="parameters"
    labelKey="parameters"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
  <Branch
    key="responses"
    labelKey="responses"
    render={[Function]}
    renderIssuesIndicators={[Function]}
  />
</ul>
`;
